engine/src/main/java/org/terasology/entitySystem/entity/internal/PojoEntityManager.java

```diff
@@ -606,29 +530,20 @@ public void saveComponent(long entityId, Component component) {
      * Implementation
      */
 
-    protected void assignToPool(EntityRef ref, EngineEntityPool pool) {
-        //Todo: job for the sector manager?
-        poolMap.put(ref.getId(), pool);
+    private EngineEntityPool getEntityPool(long id) {
+        EngineEntityPool pool = poolMap.get(id);
+        if (pool == null) {
+                logger.error("Entity {} doesn't have an assigned pool", id);
```

Comment:
At this point, should the stray entity just be assigned to the global pool? What is the behavior here if I query for the pool of a non-existing entity id? It looks like I get the global pool, which looks like the a valid value from which I can retrieve the entity as a consumer. Thus, it either
- should return null in case of error, which would lead to null checks everywhere, ...
- assign the stray entity to the global pool before returning